core/src/apps/bitcoin/verification.py

```diff
@@ -0,0 +1,104 @@
+from trezor import wire
+from trezor.crypto.hashlib import sha256
+
+from .common import ecdsa_hash_pubkey, ecdsa_verify
+from .scripts import (
+    input_script_p2wpkh_in_p2sh,
+    input_script_p2wsh_in_p2sh,
+    output_script_native_p2wpkh_or_p2wsh,
+    output_script_p2pkh,
+    output_script_p2sh,
+    parse_input_script_multisig,
+    parse_input_script_p2pkh,
+    parse_output_script_multisig,
+    parse_witness_p2wpkh,
+    parse_witness_p2wsh,
+)
+
+if False:
+    from typing import List, Tuple
+    from apps.common.coininfo import CoinInfo
+
+
+class SignatureVerifier:
+    def __init__(
+        self, script_pubkey: bytes, script_sig: bytes, witness: bytes, coin: CoinInfo,
+    ):
+        self.threshold = 1
+        self.public_keys = []  # type: List[bytes]
+        self.signatures = []  # type: List[Tuple[bytes, int]]
+        if not script_sig:
+            if len(script_pubkey) == 22:  # P2WPKH
+                public_key, signature, hash_type = parse_witness_p2wpkh(witness)
```

Comment:
The naming is correct here. The wrong variable naming is in scripts.py. In 5530aa356b48e6961296fc7d62a0766818ca2901 I fixed the variable name in the writers but forgot to do it in the readers, because I wanted that to be in a separate commit.
Done f6a2aea2c4077c9bfd86585d45bbfc8a85842cc8.

Reply:
So what is the difference between `hash_type` and `sighash_type` for you? For me `hash_type` is for example SHA256.

Reply:
It has nothing to do with the hash function used, it is the flags byte that gets appended to the signature. I explained the difference between `hash_type` and `sighash_type` in https://github.com/trezor/trezor-firmware/commit/5530aa356b48e6961296fc7d62a0766818ca2901#diff-889ef90f644349ce9441100bf4472501R499-R504.

Reply:
> It has nothing to do with the hash function used, it is the flags byte that gets appended to the signature.

That's why I find the name `hash_type` misleading. Just saying.

> hash_type and sighash_type in 5530aa3#diff-889ef90f644349ce9441100bf4472501R499-R504.

Thanks.